crates/storage/provider/src/providers/database/builder.rs

```diff
@@ -71,6 +74,29 @@ impl<N> ProviderFactoryBuilder<N> {
     ///         .unwrap();
     /// }
     /// ```
+    ///
+    /// # Open an instance with disabled read-transaction timeout
+    ///
+    /// By default, read transactions are automatically terminated after a timeout to prevent
+    /// database free list growth, which can degrade performance when the database is actively
```

Comment:
I think the main reason is to prevent db growth now, we don't have degradation due to large freelist anymore